src/rubrix/client/datasets.py

```diff
@@ -509,6 +515,8 @@ def entities_to_tuple(entities):
                 row["prediction"] = entities_to_tuple(row["prediction"])
             if row.get("annotation"):
                 row["annotation"] = entities_to_tuple(row["annotation"])
+            if row.get("metadata", "no metadata in dict") is None:
+                del row["metadata"]
```

Comment:
Could we avoid these lines by defining `metadata` as `Optional` ?

Reply:
Done